plugins/@grouparoo/mailjet/src/lib/client.ts

```diff
@@ -0,0 +1,178 @@
+  client: any;
+  apiKey: any;
+  apiSecret: any;
+
+  constructor(apiKey: string, apiSecret: string) {
+    this.apiKey = apiKey;
+    this.apiSecret = apiSecret;
+    this.client = mailjet.connect(apiKey, apiSecret);
+  }
+
+  async createList(listName: string): Promise<any> {
+    const request = await this.client
+      .post("contactslist", { version: "v3" })
+      .request({
+        Name: listName,
+      });
+    if (request?.body?.Data?.length > 0) {
+      return request?.body?.Data[0];
+    }
+    return null;
+  }
+
+  async getListByName(listName: string): Promise<any> {
+    const request = await this.client
+      .get("contactslist", { version: "v3" })
+      .request({ Name: listName });
+    if (request?.body?.Data?.length > 0) {
+      return request?.body?.Data[0];
+    }
+    return null;
+  }
+
+  async getLists(): Promise<Array<any>> {
+    const request = await this.client
+      .get("contactslist", { version: "v3" })
+      .request();
+    return request?.body?.Data;
+  }
+
+  async deleteList(listId) {
+    await this.client
+      .delete("contactslist", { version: "v3" })
+      .id(listId)
+      .request();
+  }
+
+  async addContactToList(listId, email) {
+    await this.client
+      .post("contactslist", { version: "v3" })
+      .id(listId)
+      .action("managecontact")
+      .request({
+        Action: "addnoforce",
+        Email: email,
+      });
+  }
+
+  async removeContactFromList(listId, email) {
+    await this.client
+      .post("contactslist", { version: "v3" })
+      .id(listId)
+      .action("managecontact")
+      .request({
+        Action: "remove",
+        Email: email,
+      });
+  }
+
+  async getContactListsByEmail(email: string): Promise<any> {
+    const contact = await this.getContactByEmail(email);
+    let request = null;
+    if (contact) {
+      request = await this.client
+        .get("contact", { version: "v3" })
+        .id(contact.ID)
+        .action("getcontactslists")
+        .request();
+    }
+    if (request?.body?.Data.length > 0) {
+      return request?.body?.Data;
+    }
+    return [];
+  }
+
+  async getContactByEmail(email: string): Promise<any> {
+    try {
+      const request = await this.client
+        .get("contact", { version: "v3" })
+        .id(email)
+        .request();
+      return request?.body?.Data[0];
+    } catch (e) {
+      if (e?.response?.statusCode === 404) {
+        return null;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  async deleteContactByEmail(email: string): Promise<any> {
+    const contact = await this.getContactByEmail(email);
+    if (contact) {
+      await this.deleteContact(contact.ID);
+    }
+  }
+
+  async deleteContact(contactId: string): Promise<any> {
+    const url = `https://api.mailjet.com/v4/contacts/${contactId}`;
+    await Axios({
+      method: "DELETE",
+      url,
+      auth: {
+        username: this.apiKey,
+        password: this.apiSecret,
+      },
+    });
+  }
```

Comment:
It looks like the only API that isn't in the `node-mailjet` client is DELETE... but the docs think it might be? https://github.com/mailjet/mailjet-apiv3-nodejs#delete-request.  It would be nice not to need Axios